src/Survey/ItechArt.Survey.Foundation/Authentication/Validation/Abstractions/Validator.cs

```diff
@@ -0,0 +1,24 @@
+﻿using ItechArt.Common.Validation;
+using ItechArt.Common.Validation.Abstractions;
+using ItechArt.Survey.DomainModel;
+using ItechArt.Survey.Foundation.Authentication.Abstractions;
+
+namespace ItechArt.Survey.Foundation.Authentication.Validation.Abstractions;
+
+public abstract class Validator : IValidator<User, UserRegistrationErrors>
```

Comment:
1. should be generic
2. move to common (it's base implementation)
3. use struct constraint